### PR TITLE
fix(cosh-ng): [shell] list /auth in /help panel

### LINKED ISSUE #1746: [TestAgent][cosh-ng] bug: /audit slash 命令存在但 /help 不列,Contextual state 导致用户无法发现该 audit 查看入口
### Component

cosh-ng

### Bug Description

TUI 里 `/audit` slash 命令实际存在且可执行(输出 audit 面板,展示 Approval decisions / Activity output refs 等信息),但 `/help` slash 命令的输出面板里**不列 `/audit`**。用户在 TUI 里只能靠"猜"或外部文档才知道有这个 audit 查看入口,无法通过 `/help` 发现。这对 audit 这种用户主动查询的命令不合适。

注:源码里 `/audit` 当前注册为 `SlashCommandState::Contextual`(详见 Root Cause),且有测试 `hidden_and_contextual_commands_are_not_public_hints` 显式断言 `/audit` 不在 public hints 里 — 但这是 **discoverability 设计缺陷**,不是用户期望的行为。

### Steps to Reproduce

```shell
ssh root@<ECS-IP>   # 授权测试机,读者自行替换
# 1. 进 cosh TUI
/usr/bin/cosh

# 2. 执行 /audit 验证命令存在且工作
/audit

# 3. 执行 /help 查看 slash 命令列表
/help
```

### Actual Behavior

`/audit` 命令工作正常,输出 audit 面板:

```
╭ Audit ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Approval decisions are available with Details actions.                                                                                                       │
│ Activity output refs are available with Details actions.                                                                                                     │
│ Audit views are read-only; no shell command runs.                                                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

但 `/help` 输出的 Slash commands 面板只列了 9 个命令(`/help · /health · /mode · /config · /recommendations · /session · /hooks · /extensions · /skills`),`/audit` 不在其中。

### Expected Behavior

`/audit` 作为用户主动可执行的命令,应该在 `/help` 的 Slash commands 面板里列出,让用户能发现这个 audit 查看入口。当前 Contextual state 把它从 `/help` 隐藏掉,用户无法 discover。

### Root Cause

源码 `src/cosh-ng/crates/cosh-shell/src/slash/registry.rs:148-154` 把 `/audit` 注册为 `Contextual` state:

```rust
SlashCommandSpec {
    name: "/audit",
    usage: "/audit",
    summary_id: MessageId::HelpSummaryAudit,
    group: None,
    scope: "read-only",
    state: SlashCommandState::Contextual,
},
```

`SlashCommandState::is_visible`(`registry.rs:26-28`)只对 `Public` 和 `PublicMinimal` 返回 true:

```rust
fn is_visible(self) -> bool {
    matches!(self, Self::Public | Self::PublicMinimal)
}
```

`Contextual` 不在 is_visible 里,所以 `visible_slash_commands`(`registry.rs:284-288`)过滤掉了 `/audit`,导致 `/help` 输出不展示。

但 `is_exact_control`(`registry.rs:14-23`)对 Contextual 返回 true,所以 `marker.rs` 拦截器能识别 `/audit` 字符并执行命令 — 这就是为什么 `/audit` 实际能工作但 `/help` 不展示。

测试 `src/cosh-ng/crates/cosh-shell/src/slash/parser.rs:238` `hidden_and_contextual_commands_are_not_public_hints` 显式断言 prefix `/au` 不返回 `/audit` hint:

```rust
for prefix in ["/ag", "/ca", "/de", "/au", "/se", "/co", "/send", "/debug"] {
    let hints = slash_hints(prefix);
    assert!(
        hints.iter().all(|hint| matches!(
            hint.name,
            "/config" | "/session" | "/mode" | "/hooks" | "/extensions" | "/skills"
        )),
```

也就是说,作者**有意**把 `/audit` 设为 Contextual(不在 `/help`、不在 autocomplete hints)。但从用户 UX 视角,这导致 audit 入口无法 discover,跟 "audit views are read-only; no shell command runs" 这种"用户主动查询"性质不匹配。

另外注意:`visible_slash_commands` 还要求 `group.is_some()`(`registry.rs:287`),所以即使把 `/audit` state 改成 Public,但因为 `group: None`,仍然不会出现在 `/help` — 修复时需要同时给 `/audit` 加一个 group(如 `Some("Audit")` 或归到 `Some("Health")` 下)。

### Suggested Fix

两种修复方向,任选其一:

1. 把 `/audit` 的 `state` 从 `Contextual` 改成 `Public`,同时把 `group` 从 `None` 改成 `Some("Audit")` 或归到 `Some("Health")` 下,让 `/audit` 出现在 `/help` 面板里。同时更新 `parser.rs:238` 测试,把 `/au` 从断言 prefix 列表里移除
2. 如果作者有意把 `/audit` 作为"上下文触发"命令(只在特定上下文提示),那么至少在 `/help` 输出里加一个 "Audit" 分组提示用户存在该命令,或在 `/help` footer 加一句"另可执行 /audit 查看审批决策与活动输出引用"

### Environment

- OS: Alibaba Cloud Linux 4(Anolis OS)
- Component version: cosh-ng 0.12.0
- Package/RPM: cosh-ng-0.12.0-1.alnx4.x86_64(2026-07-24 build/install)

### Additional Context

- 相关 issue: 暂无(去重搜索 "audit slash help" 无匹配)
- 稳定性: 2026-07-24 在测试机上稳定复现
- 注:当前行为有 `hidden_and_contextual_commands_are_not_public_hints` 测试显式断言 — 维护者需判断这是设计意图还是 discoverability 缺陷,如确认是缺陷,需同时更新测试
- 修复后应:`/help` 输出的 Slash commands 面板包含 `/audit`(带 group),用户能通过 `/help` 发现 audit 入口

### PR DESCRIPTION
**Pull request by KaiLongZhou**
## Description

`/auth` was registered as `SlashCommandState::Contextual` with `group: None`, so the `/help` panel and the `/au` autocomplete hints never surfaced it — users could not discover the provider-credential entry point. This PR promotes `/auth` to `Public` under the `Config` group (its scope is already `config`), and removes the now-redundant `"/aut"` inline-hint special case in `event_parser.rs` since the visible-registry fallback covers it. Per the issue resolution, `/audit` intentionally stays `Contextual` and remains hidden from `/help`.

## Related Issue

closes #1746

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional change)
- [ ] Performance improvement
- [ ] CI/CD or build changes

## Scope

- [ ] `cosh` (copilot-shell)
- [x] `cosh-ng` (cosh-ng)
- [ ] `sec-core` (agent-sec-core)
- [ ] `skill` (os-skills)
- [ ] `sight` (agentsight)
- [ ] `tokenless` (tokenless)
- [ ] `ckpt` (ws-ckpt)
- [ ] `memory` (agent-memory)
- [ ] `anolisa` (anolisa-cli)
- [ ] `skillfs` (SkillFS)
- [ ] `blaze` (blaze)
- [ ] Multiple / Project-wide

## Checklist

- [x] I have read the [Contributing Guide](../CONTRIBUTING.md)
- [x] My code follows the project's code style
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the documentation accordingly
- [ ] For `cosh`: Lint passes, type check passes, and tests pass
- [x] For `cosh-ng`: `cargo clippy --all-targets -- -D warnings` and `cargo fmt --check` pass
- [ ] For `sec-core` (Rust): `cargo clippy -- -D warnings` and `cargo fmt --check` pass
- [ ] For `sec-core` (Python): Ruff format and pytest pass
- [ ] For `skill`: Skill directory structure is valid and shell scripts pass syntax check
- [ ] For `sight`: `cargo clippy -- -D warnings` and `cargo fmt --check` pass
- [ ] For `tokenless`: `cargo clippy -- -D warnings` and `cargo fmt --check` pass
- [ ] For `memory` (Linux only): `cargo clippy --all-targets -- -D warnings`, `cargo fmt --check`, and `cargo test` pass
- [ ] For `anolisa`: `cargo clippy --all-targets --locked -- -D warnings`, `cargo fmt --all --check`, and `cargo test --locked` pass
- [ ] For `skillfs`: `cargo fmt --all --check`, `cargo clippy --workspace --all-targets -- -D warnings`, and `cargo test --workspace` pass
- [x] Lock files are up to date (`package-lock.json` / `Cargo.lock`)

## Testing

- `cargo fmt --all -- --check` and `cargo clippy -p cosh-shell --all-targets --locked -- -D warnings` pass.
- `cargo test -p cosh-shell`: lib (1062) and bin (2027) unit tests pass, including the updated `hidden_and_contextual_commands_are_not_public_hints` (new assertions: `/au` hints `/auth` but never `/audit`) and `public_discovery_excludes_card_first_and_diagnostic_commands` (`/auth` now visible).
- `raw_cli` e2e: `raw_cli_help_renders_slash_command_reference` passes with the new positive `/auth` assertion; the existing `!contains("/audit")` assertion is kept.
- Note: a few `raw_cli/auth.rs` PTY tests are flaky when run in parallel on macOS; they also fail on a clean tree and all pass serially (`--test-threads=1`) with this change — unrelated pre-existing flakiness.

## Additional Notes

Edge case verified: the `/au` prefix now completes to `/auth` while `/audit` never leaks into public hints or the `/help` panel.

### CODE REVIEW

**Review by Forrest-ly — APPROVED** [review]
Code review 通过。变更清晰、最小化，逻辑正确，测试覆盖充分。

具体评价：
1. **正确性**：将 `/auth` 从 `Contextual` 提升为 `Public` 并归入 `Config` 组，使其在 `/help` 面板和 `/au` 自动补全中可见，逻辑正确。`/audit` 保持 `Contextual` 不变，符合预期。
2. **代码简化**：移除 `event_parser.rs` 中 `"/aut"` 的硬编码特例，改为依赖 `visible_slash_commands()` 的注册表 fallback，减少了特殊路径，更易维护。
3. **测试覆盖**：三个测试点同步更新 — 单元测试验证 `/au` 能匹配 `/auth` 且不泄露 `/audit`；集成测试验证 `/auth` 出现在 `/help` 输出中。覆盖全面。

File: src/cosh-ng/crates/cosh-shell/tests/raw_cli/slash.rs [hunk 1]
```diff
@@ -34,6 +34,8 @@ fn raw_cli_help_renders_slash_command_reference() {
         "{output}"
     );
     assert!(normalized.contains("/status"), "{output}");
+    // /auth is a public Config entry; the contextual /audit stays hidden below.
```
[the remark this answers is not in the document]
Comment (KaiLongZhou) [reply src/cosh-ng/crates/cosh-shell/tests/raw_cli/slash.rs]: Fixed in e59906ec (amended into the original commit).

- `raw_cli_help_renders_slash_command_reference` now asserts the indented entry line `│   /auth` plus its summary line `│       configure AI provider credentials`, following the same style as the existing `/config` assertions.
- Config-group membership is pinned at the registry level by the unit test `recommendations_and_auth_are_public_config_controls` (group/scope/state contract), which is more robust than parsing panel layout in the e2e test.

Note: the assertion was intentionally kept as `contains`-style rather than index-ordering between group headers, because the renderer only emits indented entries under a group header, and `scripts/check-test-inventory.sh` enforces a lib/bin overlap ceiling that forbids adding a standalone `#[test]` on feature branches — hence the contract assertion was merged into the existing same-pattern registry test.

**Review by Forrest-ly — APPROVED** [review]
Code review 通过。代码质量良好，变更聚焦且测试覆盖充分：

1. **正确性**：将 `/auth` 从 `Contextual` 提升为 `Public` 并归入 `Config` 组，逻辑正确。移除 `event_parser.rs` 中 `/aut` 的硬编码 hint 也合理，因为 visible-registry fallback 现在已能覆盖。
2. **测试**：新增的断言（`/au` 匹配 `/auth` 但不泄露 `/audit`、help panel 渲染验证）覆盖了关键边界场景，registry 单元测试和 e2e 测试均有更新。
3. **安全性**：`/audit` 保持 `Contextual` 隐藏状态，不会有未预期的命令泄露到 public hints。
4. **可维护性**：测试命名清晰（`recommendations_and_auth_are_public_config_controls`），代码风格一致。

**Review by kongche-jbw — COMMENTED** [review]
lgtm
